src/evaluators/ThrowStatement.js

```diff
@@ -24,5 +25,10 @@ export default function(
 ): Value {
   let exprRef = env.evaluate(ast.argument, strictCode);
   let exprValue = Environment.GetValue(realm, exprRef);
+  if (realm.isInPureScope() && !realm.isInPureTryStatement) {
+    invariant(realm.generator !== undefined);
+    realm.generator.emitThrow(exprValue);
+    throw new PureScopeThrowCompletion(realm, ast.loc);
```

Comment:
I'm surprised and somewhat dismayed that this seems to work at all. If I understand the basic idea, this seems to be all about bubbling up the exception right to the top level, but then not emitting any code for throwing it there because the code has already been emitted here. It seems to me, therefore, that it would be safer to throw an actual throw completion here (with a flag set) and then to a) complain loudly if anything ends up catching the throw and b) not emitting any code for it at the top level.

Reply:
What if I throw a `SimpleNormalCompletion` here instead of a new type of completion? We only care for the generator entries of this execution path, not of any value thrown.

Reply:
Applied changes, now uses `SimpleNormalCompletion` and seems to work fine without needing to add another completion.